src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -135,6 +105,41 @@ internal class ConditionalTestDiscoverer
             return testCases;
         }
 
+        internal static bool EvaluateParameter(object[] conditionArguments)
+        {
+            Type calleeType = null;
+            string[] conditionMemberNames = null;
+
+            if (CheckInputToSkipExecution(conditionArguments, ref calleeType, ref conditionMemberNames)) return true;
+
+            foreach (string entry in conditionMemberNames)
+            {
+                // Null condition member names are silently tolerated
+                if (string.IsNullOrWhiteSpace(entry)) continue;
+
+                MethodInfo conditionMethodInfo = LookupConditionalMethod(calleeType, entry);
+                if (conditionMethodInfo == null)
+                {
+                    // Unable to get MethodInfo. It's caused by bad user input.
+                    // We need to report some error here. For now, just don't run the test.
+                    return false;
+                }
+
+                try
+                {
+                    bool conditionMet = (bool)conditionMethodInfo.Invoke(null, null);
+                    // If one of the conditions is false, then return the category failing trait.
+                    if (!conditionMet) return false;
+                }
+                catch (Exception exc)
+                {
+                    throw exc;
```

Comment:
Since we're rethrowing the exception I don't think we should catch it. Just remove the try catch block from here

Reply:
My original plan was to add some debug information in catch block. Since the stack trace information can be useful to identify the root cause of the problem, there is no need to rethrow the exception. Will remove the try catch block.